react/week3/fetchtodos/src/FetchTodos.js

```diff
@@ -0,0 +1,176 @@
+  const [isDone, setIsDone] = useState(false);
+  const [edit, setEdit] = useState(false);
+  const [newDescription, setNewDescription] = useState(description);
+
+  const updateTodo = () => {
+    setEdit(!edit);
+  };
+
+  return (
+    <>
+      <TodoComponent>
+        <div
+          className="list-of-todos"
+          style={
+            isDone ? { textDecoration: "line-through" } : { textDecoration: "" }
+          }
+        >
+          {edit ? (
+            <input
+              type="text"
+              value={newDescription}
+              onChange={(event) => {
+                const value = event.target.value;
+                setNewDescription(value);
+              }}
+            />
+          ) : (
+            newDescription
+          )}{" "}
+          | {deadline}
+          <div className="buttons">
+            <input type="checkbox" onClick={() => setIsDone(!isDone)} />
+            <button className="delet-button" onClick={() => removeTodo(id)}>
+              Delete
+            </button>
+            <button className="edit-button" onClick={updateTodo}>
+              {edit ? "Update" : "Edit"}
+            </button>
+          </div>
+        </div>
+      </TodoComponent>
+    </>
+  );
+}
+//This component is for input elements
+function RenderTodos() {
+  const [description, setDescription] = useState("");
+  const [deadline, setDeadline] = useState("");
+  const [apidata, setApiData] = useState([]);
+
+  const minimumDate = () => {
+    let today = new Date();
+    let dd = today.getDate();
+    let mm = today.getMonth() + 1; //January is 0!
+    let yyyy = today.getFullYear();
+    if (dd < 10) {
+      dd = "0" + dd;
+    }
+    if (mm < 10) {
+      mm = "0" + mm;
+    }
+
+    today = yyyy + "-" + mm + "-" + dd;
+    return today;
+  };
+
+  // here iam fetching the data from api
+  useEffect(() => {
+    fetch(API_URL)
+      .then((response) => response.json())
+      .then((data) => {
+        setApiData(data);
+      });
+  }, []);
+
+  //Adding the todos
+  const addTodos = (event) => {
+    event.preventDefault();
+    if (description !== "") {
+      const newTodos = {
+        id: apidata.length + 1,
+        description: description,
+        deadline: deadline,
+      };
+      setDescription("");
+      setDeadline("");
+      setApiData([...apidata, newTodos]);
+    }
+  };
+
+  //delete Todos
+  function removeTodo(id) {
+    const nextTodos = apidata.filter((todo) => todo.id !== id);
+    console.log(nextTodos);
+    setApiData(nextTodos);
+  }
+
+  return (
+    <>
+      <form onSubmit={addTodos} className="inputfields">
+        <div className="input-data">
+          <label htmlFor="tododescription">Todo Description</label>
+          <input
+            type="text"
+            placeholder="description"
+            value={description}
+            onChange={(event) => {
+              const value = event.target.value;
+              setDescription(value);
+            }}
+          />
```

Comment:
Nice with the labels!
If you want them to be more accessible, you need to add the same value in the `htmlFor` and in the `id`:
```
<label htmlFor="tododescription">Todo Description</label>
<input
  type="text"
  id="tododescription"
  ...
/>
```

Reply:
ya sure obelitz i will do that ,thank you